front/lib/api/assistant/jit_actions.ts

```diff
@@ -47,6 +57,129 @@ export async function isJITActionsEnabled(
   return use;
 }
 
+export async function listFiles(
```

Comment:
Not a big fan that this function becomes async. I think we want the snippet to live on the ContentFragmentType since we already loaded the file at some point to materialize the ContentFragmentType message right?

This will start leaking some JIT-related abstractions to the main code-base but I think it's fine since it's a simple addition of a snippet field?